tests/tensor_search/test_multimodal_tensor_combination.py

```diff
@@ -0,0 +1,604 @@
+                IndexSettingsField.index_defaults: {
+                    IndexSettingsField.model: "ViT-B/32",
+                    IndexSettingsField.treat_urls_and_pointers_as_images: True,
+                    IndexSettingsField.normalize_embeddings: False
+                }
+            })
+
+        # invalid weight string
+        tensor_search.add_documents(config=self.config, index_name=self.index_name_1, docs=[
+            {
+                "combo_text_image": {
+                    "A rider is riding a horse jumping over the barrier.": {
+                        "weight-void": 1,
+                    },
+                    "https://raw.githubusercontent.com/marqo-ai/marqo/mainline/examples/ImageSearchGuide/data/image1.jpg": {
+                        "weight": 0,
+                    },
+                },
+                "_id": "123",
+            }, ], auto_refresh=True)
+        try:
+            tensor_search.get_document_by_id(config=self.config, index_name=self.index_name_1, document_id="123")
+            raise AssertionError
+        except DocumentNotFoundError:
+            pass
+
+        # invalid weight format string
+        tensor_search.add_documents(config=self.config, index_name=self.index_name_1, docs=[
+            {
+                "combo_text_image": {
+                    "A rider is riding a horse jumping over the barrier.": {
+                        "weight-void": "1.0",
+                    },
+                    "https://raw.githubusercontent.com/marqo-ai/marqo/mainline/examples/ImageSearchGuide/data/image1.jpg": {
+                        "weight": 0,
+                    },
+                },
+                "_id": "123",
+            }, ], auto_refresh=True)
+        try:
+            tensor_search.get_document_by_id(config=self.config, index_name=self.index_name_1, document_id="123")
+            raise AssertionError
+        except DocumentNotFoundError:
+            pass
+
+        # invalid field content format
+        tensor_search.add_documents(config=self.config, index_name=self.index_name_1, docs=[
+            {
+                "combo_text_image": {
+                    "A rider is riding a horse jumping over the barrier.": 0.5,
+                    "https://raw.githubusercontent.com/marqo-ai/marqo/mainline/examples/ImageSearchGuide/data/image1.jpg": {
+                        "weight": 0,
+                    },
+                },
+                "_id": "123",
+            }, ], auto_refresh=True)
+        try:
+            tensor_search.get_document_by_id(config=self.config, index_name=self.index_name_1, document_id="123")
+            raise AssertionError
+        except DocumentNotFoundError:
+            pass
+
+    def test_validate_dict(self):
+        valid_dict = [{"test": {"weight": 0.3}, "test_test": {"weight": -0.3}},
+                      {"test": {"weight": 0.5}, "test_test": {"weight": 0.3}, "test_3": {"weight": 0.6}}]
```

Comment:
- maybe a test for ints, too: `{"weight": 2}`
- A test for {"weight": -1, "weight": 1}

Reply:
added